src/ansi-c/expr2c.cpp

```diff
@@ -3427,6 +3427,31 @@ std::string expr2ct::convert_sizeof(
   return dest;
 }
 
+/// Convert a let expression from an irep to a string, taking into account
+/// precedence.
+/// \param src: The let expression to be converted.
+/// \param precedence: The precedence to be obeyed during conversion.
+/// \return: A string representation of the expression.
+std::string expr2ct::convert_let(const exprt &src, unsigned &precedence)
+{
+  const auto &let_expr = to_let_expr(src);
+  std::string dest = "let ";
```

Comment:
??? It's unclear why this is here. Let isn't a C expression, nor is it Ada related. Why do we need this in this PR?

Reply:
(NVM turns out we do generate `let_exprt`s in Ada for some reason, need some investigation as to why)

Reply:
I think `util/format_expr` needs to know about this, but I don't think `expr2c` has any business in this?